### PR TITLE
[XRAY-2492] Update Xamarin Android

### PR DESCRIPTION
**Pull request by mduncan26**
- Improve offline storage
- Update Android Mono version

### CODE REVIEW

File: Mindscape.Raygun4Net.Xamarin.Android/RaygunClient.cs [hunk 1]
```diff
@@ -523,56 +630,136 @@ protected IEnumerable<Exception> StripWrapperExceptions(Exception exception)
...
+     
+      bool canSend = OnSendingMessage(raygunMessage);
+
+      if (!canSend)
+      {
+        RaygunLogger.Debug("Sending message cancelled");
+        return;
+      }
+
+      var internet = HasInternetConnection;
```
Comment (JasenPalmer) [inline Mindscape.Raygun4Net.Xamarin.Android/RaygunClient.cs]: Looks like this variable is unused?

File: Mindscape.Raygun4Net.Xamarin.Android/RaygunFileManager.cs [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+      try
+      {
+        // Get the paths to all the files in the crash report directory.
+        var files = Directory.GetFiles(GetCrashReportDirectory());
+
+        foreach (var filePath in files)
+        {
+          // Read the content of the file on disk.
+          var raygunFile = ReadCrashReportFromDisk(filePath);
```
Comment (JasenPalmer) [inline Mindscape.Raygun4Net.Xamarin.Android/RaygunFileManager.cs]: `filePath` implies a path to a file but the method `ReadCrashReportingFromDisk` has `fileName` as its parameter. Just a little inconsistent

File: Mindscape.Raygun4Net.Xamarin.Android/RaygunResponseStatusCode.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  }
+
+  public class RaygunResponseStatusCodeConverter
+  {
+    public static string ToString(int statusCode)
+    {
+      switch (statusCode)
+      {
+        case (int)RaygunResponseStatusCode.Accepted: 
+        return "Request succeeded";
```
Comment (QuantumNightmare) [inline Mindscape.Raygun4Net.Xamarin.Android/RaygunResponseStatusCode.cs]: I'm used to seeing indenting on these, any reason not to?

File: Mindscape.Raygun4Net.Xamarin.Android/RaygunResponseStatusCode.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
+﻿using System;
```
Comment (QuantumNightmare) [inline Mindscape.Raygun4Net.Xamarin.Android/RaygunResponseStatusCode.cs]: nitpick: new line between usings and namespace

File: Mindscape.Raygun4Net.Xamarin.Android/RaygunResponseStatusCode.cs (hunk 3 above)
Comment (mduncan26) [reply Mindscape.Raygun4Net.Xamarin.Android/RaygunResponseStatusCode.cs]: Just auto formatted by VS. Ive indented them now.

File: Mindscape.Raygun4Net.Xamarin.Android/RaygunFileManager.cs (hunk 2 above)
Comment (mduncan26) [reply Mindscape.Raygun4Net.Xamarin.Android/RaygunFileManager.cs]: Agreed, fixed up.

File: Mindscape.Raygun4Net.Xamarin.Android/RaygunClient.cs (hunk 1 above)
Comment (mduncan26) [reply Mindscape.Raygun4Net.Xamarin.Android/RaygunClient.cs]: Oops from testing. Removed it.

File: Mindscape.Raygun4Net.Xamarin.Android/RaygunResponseStatusCode.cs (hunk 4 above)
Comment (mduncan26) [reply Mindscape.Raygun4Net.Xamarin.Android/RaygunResponseStatusCode.cs]: Agreed, the system reference wasn't needed in the end too.

**Pull request merged**